### PR TITLE
Feature/user profile main form

### PR DESCRIPTION
**Pull request by aussie-m-mike**

### CODE REVIEW

File: client/src/components/profile/ProfileForm.vue [hunk 1]
```diff
@@ -1,9 +1,216 @@
...
+            dateFormat: 'rcf-1123',
+            timeFormat: '12h',
+            country: '',
+            timeZone: null
+        },
+
+        languages: [enLang.ENGLISH, enLang.GERMAN, enLang.UKRAINIAN],
+        dateFormats: [
+            { value: 'rcf-1123', text: 'Mon, 10 Aug 2020 14:38:15 +0000' },
+            { utc: 'utc', text: '2020-08-10T14:38:15Z' },
```
Comment (lenchvolodymyr) [inline client/src/components/profile/ProfileForm.vue:144]: why key is 'utc'?

Comment (lenchvolodymyr) [reply client/src/components/profile/ProfileForm.vue:144]: text is better in human readable format as on clendly yyyy-mm-dd

File: client/src/store/modules/profile/actions.js [hunk 2]
```diff
@@ -21,11 +25,21 @@ export default {
...
+        commit(UPDATE_AVATAR, url);
+
         return response?.data?.data;
     },
 
-    async updateProfile() {}
+    async updateProfile({ commit }, profile) {
+        const response = profileService.updateProfile(profile);
+
+        commit(UPDATE_PROFILE, profile);
```
Comment (lenchvolodymyr) [inline client/src/store/modules/profile/actions.js:41]: can't find the mutator, you should update a user

File: client/src/components/profile/ProfileForm.vue [hunk 3]
```diff
@@ -1,9 +1,216 @@
...
+            this.file = event.target.files[0];
+            this.newAvatar = URL.createObjectURL(this.file);
+        },
+
+        async onSaveHandle() {
+            try {
+                if (this.avatarIsNew) {
+                    const response = await this.updateAvatar(this.avatar);
+                    const avatar = response?.avatar?.url;
+                    this.userProfile = avatar;
```
Comment (lenchvolodymyr) [inline client/src/components/profile/ProfileForm.vue:186]: why do you replace whole userProfile with avatar?

Comment (lenchvolodymyr) [reply client/src/components/profile/ProfileForm.vue:186]: flow must be, to upload file, then update whole profile and change global state.

**Comment by aussie-m-mike** [comment]
Made the requested changes and also suggestions from the DM

**Pull request merged**